scapy/contrib/automotive/uds_scan.py

```diff
@@ -320,6 +367,156 @@ def _get_table_entry(tup):
                 label)
 
 
+class UDS_RDBISelectiveEnumerator(StagedAutomotiveTestCase):
+    @staticmethod
+    def __connector_random_to_sequential(rdbi_random, _):
+        # The scan range for the sequential enumerator
+        # is generated in the sequential enumerator itself,
+        # since in this connector we do not know the state
+        # of the ECU.
+        return {"rdbi_random": rdbi_random}
+
+    def __init__(self):
+        # type: () -> None
+        super(UDS_RDBISelectiveEnumerator, self).__init__(
+            [UDS_RDBIRandomEnumerator(), UDS_RDBIEnumerator()],
+            [None, self.__connector_random_to_sequential])
+
+
+class UDS_RDBIRandomEnumerator(UDS_RDBIEnumerator):
+    block_size = 2 ** 6
+
+    def _get_initial_requests(self, **kwargs):
+        # type: (Any) -> Iterable[Packet]
+
+        probabilities = [0.015625, 0.0, 0.0, 0.0, 0.453125, 0.34375, 0.296875, 0.0, 0.171875, 0.171875, 0.203125,
```

Comment:
Is there a way to simplify this probabilities? In this style, they are not that maintainable. Could you please simplify them and make a comment how they are used.

Reply:
Do you have an idea of how to simplify them?

Reply:
Not at the moment. But I would prefer something simpler

Reply:
You could compress zeros like ` 0.123] + [0] * 10 + [0.33......`

Reply:
can you lower the resolution, to maybe combine more probabilities
or somehow round the values -- > 0.0156 to 2   ( if you multiply by 100 and ceil)

Reply:
Changed